File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Visitor/Module.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Visitor {
    internal class Module : IVisited {
        private readonly List<Section> sections = new List<Section>();

        public void AcceptVisitor(IVisitor v) {
            v.VisitModuleStart(this);
            foreach (var s in sections) {
                s.AcceptVisitor(v);
            }
            v.VisitModuleEnd();
        }

        public void AddSection(Section s) {
            sections.Add(s);
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/visitor/Module.java
package patterns.visitor;

import java.util.*;

public class Module implements Visited {
	private List<Section> sections = new ArrayList<Section>();
	 
	public void acceptVisitor(Visitor v) {
		v.visitModuleStart(this);
		for(Section s : sections) {
			s.acceptVisitor(v);
		}
		v.visitModuleEnd();
	}
	public Module() {
		super();
	}
	public void addSection(Section s) {
		sections.add(s);
	}
}
